Add explanations to 7 9 (#20)
MassimiIiano/software-eng-projects-uni: README.md

![Klassendiagramm mit Funktionen](./sw-eng-07.drawio.svg)

#### Erklärung

> Dieses diagramm folgt den data flow. Die Presentation componente zeigen die Suchergebnisse, Buchungsvorshläge und Buchungsvorgänge ab. diese daten kriegen sie vom controll layer der auf einem eigenen server leuft. dieser ist dafür zuständig die korrekten informationen vom database zu holen. Wichtig zu bemerken dass die einzige Entität die nich in ein database gespeicehrt wird die EVorschlag Entität ist. dieese wird dinamisch von CVorschlag anhand von den verfügbaren EAutos berechnet. Falls ein buchungsvorschlag ausgewählt wird wird ein Buchungsobjekt anhand von diesen EVorschlag gemacht, der dan von CBuchung im database abgespeihert wird.

Reviewer: ```suggestion
Dieses Diagramm folgt dem Datenfluss. Die Präsentationskomponente zeigt die Suchergebnisse, Buchungsvorschläge und Buchungsvorgänge an. Diese Daten erhält sie vom Kontroll-Layer, der auf einem eigenen Server läuft und dafür zuständig ist, die korrekten Informationen aus der Datenbank abzurufen. Wichtig ist zu beachten, dass die einzige Entität, die nicht in der Datenbank gespeichert wird, die Entität *EVorschlag* ist. Diese wird dynamisch von *CVorschlag* anhand der verfügbaren *EFahrzeug* berechnet. Falls ein Buchungsvorschlag ausgewählt wird, wird ein *Buchungsobjekt* anhand dieses *EVorschlags* erstellt, welches dann von *CBuchung* in der Datenbank gespeichert wird.
```

Reviewer 2: These go on top of Nick's suggestion:

```diff
-Die Präsentationskomponente
+Die Präsentations-Layer
```

```diff
-Diese Daten erhält
+Die Daten werden vom Kontroll-Layer bezogen
```

```diff
-Wichtig ist zu beachten, dass die einzige Entität, die nicht in der Datenbank gespeichert wird, die Entität *EVorschlag* ist
+Wichtig ist zu beachten, dass die einzige Entität, die nicht in der Datenbank gespeichert wird, EVorschlag ist.
```

```diff
-Diese wird dynamisch von *CVorschlag* anhand der verfügbaren *EFahrzeug* berechnet.
+EVorschlag wird dynamisch von CVorschlag auf der Grundlage der verfügbaren EFahrzeuge berechnet.
```

```diff
-welches dann von *CBuchung* in der Datenbank gespeichert wird
+das dann von *CBuchung* in der Datenbank gespeichert wird
```

### A9: Komponentendiagramm
![Klassendiagramm mit Funktionen](./sw-eng-09.drawio.svg)

> Dieses Komponentendiagramm zeigt die Schichtenarchitektur eines Autovermietungssystems mit drei Hauptschichten:

Reviewer: ```suggestion
Dieses Komponentendiagramm zeigt die Schichtenarchitektur eines Autovermietungssystems mit drei Hauptschichten:

AVEntities (Datenbank-/Entitäten-Schicht): Speichert die Kernentitäten des Systems, wie z. B. Fahrzeug, Kunde, Buchung, Vorschlag, Extra und Bezahlung.

AVController (Geschäftslogik-Schicht): Verarbeitet die Hauptfunktionen des Systems, wie Suche, Buchung, Vorschlag und Mail. Diese Schicht verbindet die Datenbank mit der Benutzeroberfläche. Für den Kunden relevante Daten, wie z. B. Buchungsdaten, werden ihm automatisch per E-Mail zugesendet.

AVPresentation (Präsentationsschicht): Die Benutzeroberfläche, die dem Nutzer ermöglicht, mit den Funktionen wie Suche, Buchung und Vorschlag zu interagieren.

Die Schichten sind so organisiert, dass jede Schicht ihre spezifische Aufgabe hat, was die Wartbarkeit und Skalierbarkeit des Systems fördert
```

> 

Reviewer: ```suggestion

```
